src/complex/Utilities/Parsing/HDF5/H5DatasetWriter.hpp

```diff
@@ -185,19 +178,18 @@ class COMPLEX_EXPORT DatasetWriter : public ObjectWriter
    */
   void createOrOpenDataset(H5::IdType typeId, H5::IdType dataspaceId);
 
+  /**
+   * @brief Closes the HDF5 dataset and resets the ID to 0.
+   */
+  virtual void closeHdf5() override;
```

Comment:
```suggestion
  void closeHdf5() override;
```
You do not need virtual and override on the same method.